median/plugin.go

```diff
@@ -39,6 +40,14 @@ func (p *Plugin) NewMedianFactory(ctx context.Context, provider types.MedianProv
 		OnchainConfigCodec: provider.OnchainConfigCodec(),
 		ReportCodec:        provider.ReportCodec(),
 	}
+	if cr := provider.ChainReader(); cr != nil {
+		factory.ContractTransmitter = &medianContract{cr, types.BoundContract{
+			//TODO address? but that should be implied by the provider already
```

Comment:
@reductionista What is the correct way to handle this in the short-term?

Reply:
Ryan is in the process of refactoring this. I think we've decided we're going to change the signature of `GetLatestValue` to only require passing `contractName string` instead of `contract BoundCountract`, as the relay should be able to look up the address from the contract name in its config.

We haven't decided on what to do about the `Pending` field of `BoundContract` yet though:  if there's a need for products to be able to dynamically change whether "latest value" means "as of the latest finalized block" or "as of the latest unfinalized block", then it needs to be a separate param. If we can assume each product will always want one or the other we can just have it in the RelayConfig section of the job spec.

I'd be tempted to say "let's not add it until a product says they need it", because it would be annoying for every product to have to pass `false` as an extra explicit param to `GetLatestValue()`.  On the other hand, changing the signature of methods in the `chainlink-commons` API is a pain and probably best done earlier rather than later.  Any thoughts on this?

Reply:
Oh, actually I know the solution to this:  we'll just replace it with `contractName` and not have pending flag.  Separately, we'll add a new method at some point called `UpdateRelayConfig` so that the product can make any changes it wants to the relay config after the job is already running.

This will solve the Pending flag issue. But more importantly, it's needed anyway for the address.  Functions was the reason why we decided to pass the address to `GetLatestValue()` originally rather than just having it hard coded in `RelayConfig`:  they allow arbitrary updates to the contract address they're interacting with while the job is running.  We don't have to add this now, since neither of these features are in use yet, but probably next quarter.

Reply:
> because it would be annoying for every product to have to pass false as an extra explicit param to GetLatestValue().

The point of `BoundContract` was to prevent this and to give us a place to add new fields without breaking the signature.

Reply:
> UpdateRelayConfig so that the product can make any changes it wants to the relay config after the job is already running.

Introducing mutable config seems like it might be overkill for solving this problem. How would that even work?

Reply:
I guess we might want to name `UpdateRelayConfig` something more specific, since product plugins are not supposed to understand the format of RelayConfig.  This should only be used for things which apply in the same way to all chains but the relay needs to know about.  Presently, I think the only two things which fall into this category are the mapping between contract names and addresses, and this pending flag.  But there could be more things in the future I imagine?

Reply:
> Introducing mutable config seems like it might be overkill for solving this problem. How would that even work?

The problem of the relay understanding how contract names map to addresses is a pretty complex one. I hadn't realized the full complexity of it until a recent meeting @nolag and I had on it where we thought through different options for how to initialize the relay.
On Solana, a single chain-agnostic contract name (in this case "median") will map to multiple chain-specific contract names & addresses, so the BoundContract struct we presently have already won't work on that chain.  For the most part, this mapping will remain fixed and would likely apply to more than just `GetLatestValue` in the future, which makes it naturally something which belongs in a general config of some type.  However, because of the specific ask from Functions (which has to do with a special dynamic way of upgrading a deployed contract they're in the process of implementing) this must be mutable.  So I think mutable config makes more sense in this case than passing it every time to `GetLatestValue` and any other similar methods we implement in the future.  They can all share the same config, which is also used at initialization.

Come to think of it though, we might want to decouple this more from the `RelayConfig` struct we presently have which is based on the job spec.  It seems like this is more like `PluginConfig` since it's something plugin-specific... it's just something that's semi-permanent that the relay needs to be aware of throughout its lifecycle.  Can we put these in `PluginConfig` and have both of those passed to the relay?  I think we do that for several other relay services while starting them up.

That said, having some sort of flexible option you can pass to `GetLatestValue` does make sense. It's something @ilija42 and I discussed a while back as a possibility when thinking about the unused `lookback` param libocr includes in the signature of `LatestRoundRequested` .  We ended up deciding it's not needed for `lookback`, and research confirmed that this parameter can be safely ignored on all chains (I think it was originally there for LogBroadcaster, but is no longer used anywhere).  However, the possibility remains that a plugin might need to pass additional params in the future for some reason.  Maybe we should just have some kind of catch-all `opts` param at the end?

Reply:
Is 'mutating' the right word? IMHO we should not be mutating anything derived from the job spec config. On the other hand, I don't see any problem with being able to dynamic add to the configuration, but I wouldn't call that mutating per se.

Reply:
Maybe in the long run what we need is for there to be 3 types of config:  plugin-only, relay-only, and shared config... and shared config would have the same format for all relays and plugins but each plugin could set it differently before passing it to the relay?  My sense is that we already have needs for shared config in several places, but we've got various workarounds in different plugins like passing PluginConfig down to the relay which are substituting for that.

Reply:
Anyway, for the very short term (merging this PR) I think we can leave it as-is and just not fill in the address